saw-remote-api/python/tests/saw/test_ghost.py

```diff
@@ -0,0 +1,65 @@
+import saw
+from saw.llvm import Contract, CryptolTerm, cryptol, void, i32, GhostVariable
+
+import unittest
+from pathlib import Path
+
+def pre_counter(contract: Contract, counter: GhostVariable):
+    n = contract.fresh_var(i32, "n")
+    contract.precondition(n < cryptol("128"))
+    contract.ghost_value(counter, n)
+    return n
+
+def post_counter(contract: Contract, counter: GhostVariable, n: CryptolTerm):
+    contract.ghost_value(counter, cryptol("(+)")(n, cryptol("1")))
+
+class GetAndIncrementContract(Contract):
+    def __init__(self, counter: str) -> None:
+        super().__init__()
+        self.counter = counter
+
+    def specification(self) -> None:
+        n = pre_counter(self, self.counter)
+        self.execute_func()
+        post_counter(self, self.counter, n)
+        self.returns(n)
+
+class FContract(Contract):
+    def __init__(self, counter: str) -> None:
+        super().__init__()
+        self.counter = counter
+
+    def specification(self) -> None:
+        n = pre_counter(self, self.counter)
+        i = self.fresh_var(i32, "i")
+        self.precondition(i < cryptol("512"))
+        self.execute_func(i)
+        post_counter(self, self.counter, n)
+        self.returns(cryptol("(*)")(i, n))
+
+class GhostTest(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(self):
+        saw.connect(reset_server=True)
+
+    @classmethod
+    def tearDownClass(self):
```

Comment:
Be aware that @pnwamk is replacing this `setUpClass`/`tearDownClass` two-step tango with a single call to `connect(reset_server=True)` in #1174. I'm not sure which of these two PRs will land first. If this lands first, then no further action will be required here, but if #1174 lands first, it would be good to use the same connection-related conventions.

Reply:
Ah, good to know. I think this one will land first, as @pnwamk wasn't sure about merging the other one right before the release.

Reply:
The other PR ended up not really changing the _server_ at all and has already landed (it was sufficient to just add some small pauses to the tests so they won't step on eachother's toes). As long as the CI passes I don't think it's a big deal if this file uses `setUpClass` etc.

Reply:
Easy enough to change. I'll do it.